framework/src/Jovo.ts

```diff
@@ -65,17 +66,19 @@ export interface JovoPersistableData {
 export interface JovoComponentInfo<
   DATA extends ComponentData = ComponentData,
   CONFIG extends UnknownObject = UnknownObject,
+  EVENTS extends string = string,
 > {
   data: DATA;
   config?: CONFIG;
+  resolve?: Record<EVENTS, string | ((this: BaseComponent, ...args: any[]) => any)>;
 }
 
 export interface DelegateOptions<
   CONFIG extends UnknownObject | undefined = UnknownObject | undefined,
   EVENTS extends string = string,
 > {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  resolve: Record<EVENTS, string | ((this: BaseComponent, ...args: any[]) => any)>;
+  resolve: Partial<Record<EVENTS, string | ((this: BaseComponent, ...args: any[]) => any)>>;
```

Comment:
`resolve` object now accepts undefined values in order to not require all events (due to how Record<EVENTS, ...> behave).